src/wasm/wat-parser.cpp

```diff
@@ -925,18 +947,61 @@ template<typename Ctx> MaybeResult<> strtype(Ctx& ctx, ParseInput& in) {
   }
 
   if constexpr (parsingDecls<Ctx>) {
-    ctx.typeDefs.push_back({name, start});
+    return name;
+  } else {
+    return Ok{};
   }
-  return Ok{};
 }
 
 // subtype ::= '(' 'sub' typeidx? strtype ')'
 //           | strtype
```

Comment:
I wasn't familiar with this notation of `(sub ..)`. Which spec is that in?

Reply:
This is how subtypes are declared in the upstream GC spec: https://github.com/WebAssembly/gc/blob/main/test/core/gc/type-subtyping.wast. Now that I'm actually looking at the examples, though, I see that I have the order slightly wrong and that the `sub` should be inside the `type`. I'll fix the ordering.